library/gui/screen/src/main/java/org/quiltmc/qsl/screen/mixin/client/MinecraftClientMixin.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2016, 2017, 2018, 2019 FabricMC
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.screen.mixin.client;
+
+import org.jetbrains.annotations.Nullable;
+import org.quiltmc.qsl.screen.api.client.ScreenEvents;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.screen.Screen;
+
+@Mixin(MinecraftClient.class)
+abstract class MinecraftClientMixin {
+	@Shadow
+	public Screen currentScreen;
+
+	@Unique
+	private Screen quilt$tickingScreen;
+
+	@Inject(method = "setScreen", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/screen/Screen;removed()V", shift = At.Shift.AFTER))
+	private void onScreenRemove(@Nullable Screen screen, CallbackInfo ci) {
+		ScreenEvents.remove(this.currentScreen).invoker().onRemove(this.currentScreen);
+	}
+
+	@Inject(method = "stop", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/screen/Screen;removed()V", shift = At.Shift.AFTER))
+	private void onScreenRemoveBecauseStopping(CallbackInfo ci) {
+		ScreenEvents.remove(this.currentScreen).invoker().onRemove(this.currentScreen);
+	}
+
+	// Synthetic method in `tick`
+	// These two injections should be caught by "Screen#wrapScreenError" if anything fails in an event and then rethrown in the crash report
+	@Inject(method = "method_1572", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/screen/Screen;tick()V"))
```

Comment:
Same thing in this class; some comments explaining how to find the lambda if the intermediary name changes would be nice

Reply:
Same for this

Reply:
Same for this

Reply:
Fixed.